test/NugetFeedDeploymentRepositoryTest/NugetFeedApplicationRepositoryTest.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Etg.Yams.Application;
+using Etg.Yams.NuGet.Storage;
+using Etg.Yams.Storage;
+using Etg.Yams.Utils;
+using Xunit;
+
+namespace Etg.Yams.Nuget.Storage.Test
+{
+    [Trait("Category", "Integration")]
+    public class NugetFeedDeploymentRepositoryTest
+    {
+        [Fact]
+        public async Task TestHasBinaries()
+        {
+            NugetFeedApplicationRepository _applicationRepository = new NugetFeedApplicationRepository();
+            Assert.True(await _applicationRepository.HasApplicationBinaries(new AppIdentity("jQuery", "3.3.1")));
+        }
+
+        [Fact]
+        public async Task TestHasBinaries_VersionDoesntExist()
+        {
+            NugetFeedApplicationRepository _applicationRepository = new NugetFeedApplicationRepository();
+            Assert.False(await _applicationRepository.HasApplicationBinaries(new AppIdentity("jQuery", "0.0.0-thisversiondoesntexist")));
+        }
+
+        [Fact]
+        public async Task TestHasBinaries_PackageDoesntExist()
+        {
+            NugetFeedApplicationRepository _applicationRepository = new NugetFeedApplicationRepository();
+            Assert.False(await _applicationRepository.HasApplicationBinaries(new AppIdentity(Guid.NewGuid().ToString(), "1.0.0")));
+        }
+
+        [Fact]
+        public async Task TestDownloadApplicationBinaries()
+        {
+            string localPath = await CreateTestTempDirectory("Nuget_TestDownloadApplicationBinaries");
+            NugetFeedApplicationRepository _applicationRepository = new NugetFeedApplicationRepository();
+            await _applicationRepository.DownloadApplicationBinaries(new AppIdentity("jQuery", "3.3.1"), localPath, ConflictResolutionMode.OverwriteExistingBinaries);
+
+            var files = FileUtils.ListFilesRecursively(localPath);
+
+            Assert.Contains(Path.Combine(localPath, "Content", "Scripts", "jquery-3.3.1.min.js"), files);
+        }
+
+        [Fact]
+        public async Task TestDownloadApplicationBinaries_ThrowsWhenPackageNotFound()
+        {
+            AppIdentity appIdentity = new AppIdentity(Guid.NewGuid().ToString(), "1.0.0");
+            string localPath = await CreateTestTempDirectory("Nuget_TestDownloadApplicationBinaries");
+            NugetFeedApplicationRepository _applicationRepository = new NugetFeedApplicationRepository();
+
+            BinariesNotFoundException exception = await Assert.ThrowsAsync<BinariesNotFoundException>(() => _applicationRepository.DownloadApplicationBinaries(appIdentity, localPath, ConflictResolutionMode.OverwriteExistingBinaries));
+            Assert.Contains(appIdentity.ToString(), exception.Message);
```

Comment:
Testing the exception message feels a bit fragile

Reply:
I want to ensure that the message contains app identity. I don't validate the whole message, just the important piece of information. I can remove this assertion if you insist.

Reply:
I don't insist :) sounds reasonable.